### PR TITLE
UPSTREAM: 111306: Make scheduling e2e tests run PSa-restricted pods

### PR DESCRIPTION
**Pull request by stlaz**
The "pause" pods that are being run in the scheduling tests are
sometimes launched in system namespaces. Therefore even if a test
is considered to be running on a "baseline" Pod Security admission
level, its "baseline" pods would fail to run if the global PSa
enforcement policy is set to "restricted" - the system namespaces
have no PSa labels.

The "pause" pods run by this test can actually easily run with
"restricted" security context, and so this patch turns them
into just that.

/assign @ingvagabund

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/approve
/remove-label backports/unvalidated-commits
/label backports/validated-commits
/hold
until we get upstream approved

**Comment by stlaz** [comment]
/unhold
upstream approved \o/

**Pull request merged**
